patroni/scripts/barman_recover.py

```diff
@@ -0,0 +1,486 @@
+#!/usr/bin/python3
+
+"""Restore a Barman backup to the local node through ``pg-backup-api``.
+
+This script can be used both as a custom bootstrap method, and as a custom
+create replica method. Check the output of ``--help`` to understand the
+parameters supported by the script. ``--datadir`` is a special parameter and it
+is automatically filled by Patroni in both cases.
+
+It requires that you have previously configured a Barman server, and that you
+have ``pg-backup-api`` configured and running in the same host as Barman.
+
+Refer to :class:`ExitCode` for possible exit codes of this script.
+"""
+from argparse import ArgumentParser
+from enum import IntEnum
+import json
+import logging
+import ssl
+import sys
+import time
+from typing import Any, Callable, Optional, Tuple, Type, Union, TYPE_CHECKING
+from urllib.error import HTTPError, URLError
+from urllib.parse import urljoin
+from urllib.request import Request, urlopen
```

Comment:
Hmm... Why not just `urllib3`?

Reply:
Hmm, I tried to avoid additional dependencies because initially I didn't think of shipping this as part of Patroni as an application build by `setup.py`.
I can take a look at using `urllib3` instead of `urllib` as Patroni already requires the former.

Reply:
https://github.com/zalando/patroni/blob/ce10e5fccc8eb6fc3e0e6bd177f6958d463904fd/setup.py#L26-L28

Everything that is not in EXTRAS_REQUIRE is a direct dependency: https://github.com/zalando/patroni/blob/ce10e5fccc8eb6fc3e0e6bd177f6958d463904fd/requirements.txt#L1

Reply:
Yeah, exactly :)